lib/jsonwp-proxy/proxy.js

```diff
@@ -155,19 +158,94 @@ class JWProxy {
     return [res, resBody];
   }
 
+  /**
+   * W3C /timeouts can take as many as 3 timeout types at once, MJSONWP /timeouts only takes one
+   * at a time. So if we're using W3C and proxying to MJSONWP and there's more than one timeout type
+   * provided in the request, we need to do 3 proxies and combine the result
+   *
+   * @param {Array} timeoutObjects List of MJSONWP + W3C compatible timeouts request objects
+   */
+  getTimeoutRequestObjects (body) {
+    const {protocol} = body;
+    const timeoutObjects = [];
+    const baseObj = {
+      script: body.script,
+      pageLoad: body.pageLoad,
+      implicit: body.implicit,
+    };
+    if (protocol === W3C) {
+      for (let [type, ms] of _.toPairs(baseObj)) {
+        if (_.isNumber(ms)) {
```

Comment:
should we also check if type is defined?

Reply:
`type` will always be defined because it's the key of an object